cc/README.md

```diff
@@ -9,115 +10,188 @@ build system. We have experienced build issues with the latest version of Bazel,
-  --linkopt=-fsanitize=address \
-  --copt=-fno-omit-frame-pointer \
-  --copt=-O1
+bazel build -c opt cc:simple_example
+bazel-bin/cc/simple_example \
+  --model=saved_models/000990-cormorant.pb \
+  --num_readouts=160
 ```
 
-## Running self play:
+#### cc:selfplay
+
+The self-play binary used in our training pipeline. Has a lot more functionality
+that the simple example, including:
 
-The C++ Minigo binary requires the model to be provided in GraphDef binary
-proto format.
+ - Play multiple games in parallel, batching their inferences together for
+   better GPU/TPU utilization.
+ - Automatic loading of the latest trained model.
+ - Write SGF games & TensorFlow training examples to Cloud Storage or Cloud
+   BigTable.
+ - Flag file support with automatic reloading. This is used among other things
+   to dynamically adjust the resign threshold to minimize the number of bad
+   resigns.
 
-To run Minigo with a 9x9 model:
 
 ```shell
-bazel build --define=board_size=9 -c opt cc:main
-bazel-bin/cc/main --model=$MODEL_PATH --mode=selfplay
+bazel build -c opt cc:selfplay
+bazel-bin/cc/selfplay \
+  --model=saved_models/000990-cormorant.pb \
+  --num_readouts=160 \
+  --parallel_games=1 \
+  --output_dir=data/selfplay \
+  --holdout_dir=data/holdout \
+  --sgf_dir=sgf
 ```
 
-To run Minigo with a 19x19 model:
+#### cc:eval
+
+Evaluates the performance of two models by playing them against each other over
+multiple games in parallel.
 
 ```shell
-bazel build -c opt cc:main
-bazel-bin/cc/main --model=$MODEL_PATH --mode=selfplay
+bazel build -c opt cc:eval
+bazel-bin/cc/eval \
+  --model=saved_models/000990-cormorant.pb \
+  --model_two=saved_models/000990-cormorant.pb \
+  --num_readouts=160 \
+  --parallel_games=32 \
+  --sgf_dir=sgf
 ```
 
-The Minigo binary has a lot of command line arguments that configure its
-behavior, run it with --helpshort to see the full list.
+#### cc:gtp
+
+Play using the GTP protocol. This is also the binary we recommend using as a
+backend for Minigui (see `minigui/README.md`).
 
 ```shell
-bazel-bin/cc/main --helpshort
+bazel build -c opt cc:gtp
+bazel-bin/cc/gtp \
+  --model=saved_models/000990-cormorant.pb \
+  --num_readouts=160
 ```
 
-## Design
+#### cc:puzzle
 
-The general structure of the C++ code tries to follow the Python code where
-appropriate, however a lot of the implementation details are different for
-performance reasons. In particular, the logic that handles the board state
-strives for a small memory footprint and eschews data structures more
-sophisticated than a simple array in an effort to minimize memory allocations
-and maximize cache locality of data. In doing so, the implementation is kept
-fairly simple (there is no need for a LibertyTracker) and at the time of writing
-performance of the Position code was more than 450x that of its Python
-counterpart.
+Loads all SGF files found in the given `sgf_dir` and tries to predict the move
+made at each position in each game.
```

Comment:
what's the output format? Does it edit the sgfs or output a new directory of sgfs with the model's guesses?

Reply:
I didn't write this code. It looks like it just prints the results to stderr.